raiden/api/rest.py

```diff
@@ -257,6 +258,16 @@ def open(self, partner_address, token_address, settle_timeout, balance=None):
             )
         except (InvalidAddress, InvalidSettleTimeout, SamePeerAddress) as e:
             return make_response(str(e), httplib.CONFLICT)
+        except ValueError as e:
+            if re.match("Token address 0x[0-9a-f]{40} does not contain code", str(e)):
```

Comment:
I've just adapted what is in the `network/rpc/client.py` already - it is using ValueError for all exceptions. We can change this to something more sensible later, but I wouldn't change pattern the class is using as it may break other exception handlers.